e2e-tests/IIoTPlatform-E2E-Tests/deploy/IIoTHubDeployment.cs

```diff
@@ -0,0 +1,22 @@
+﻿// ------------------------------------------------------------
+//  Copyright (c) Microsoft Corporation.  All rights reserved.
+//  Licensed under the MIT License (MIT). See License.txt in the repo root for license information.
+// ------------------------------------------------------------
+
+namespace IIoTPlatform_E2E_Tests.TestExtensions {
+    using System.Collections.Generic;
+    using System.Threading.Tasks;
+
+    public interface IIoTHubDeployment {
+
+        /// <summary>
+        /// Create a new layered deployment or update an existing one
+        /// </summary>
+        Task<bool> CreateOrUpdateLayeredDeploymentAsync();
+
+        /// <summary>
+        /// Create a deployment modules object
+        /// </summary>
+        IDictionary<string, IDictionary<string, object>> CreateDeploymentModules();
```

Comment:
Judging by the usage of this method in `IoTHubEdgeBaseDeployment` and `IoTHubPublisherDeployment` classes, it seems like it is more of a way to separating responsibilities (of generating deployment content) rather than a part necessary part of public interface. The method itself is not called from outside those classes. So I would argue that  `CreateDeploymentModules()` method should not be a part of `IIoTHubDeployment` interface.